src/helpers.rs

```diff
@@ -510,9 +509,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, (bool, u64)> {
         let target_os = self.eval_context_ref().tcx.sess.target.target.target_os.as_str();
         match target_os {
-            "linux" => self.write_os_str_to_c_str(os_str, mplace.ptr, size),
+            "linux" | "macos" => self.write_os_str_to_c_str(os_str, mplace.ptr, size),
             "windows" => self.write_os_str_to_wide_str(os_str, mplace, size),
-            _ => panic!("support for target OS not yet available"),
+            _ => throw_unsup_format!("support for target OS not yet available"),
```

Comment:
```suggestion
            _ => throw_unsup_format!("OsString support for target OS not yet available"),
```
Would be better if the message that support for *what* is not available.